### PR TITLE
Make ByteView constructible from arrays. Kill full_view

### PR DESCRIPTION
**Pull request by yperbasis**
Reincarnation of PR #427.

Also `upsert_storage_value` now trims left zeros from value on itself.

### CODE REVIEW

**Review by greg7mdp — COMMENTED** [review]
Looks great!

**Comment by yperbasis** [comment]
Putting this PR on hold until PR #505 is merged.

File: core/silkworm/common/util.cpp
Comment (AndreaLanfranchi) [inline core/silkworm/common/util.cpp:126]: data is not consumed nor altered.
I'd have it passed by const ref

File: core/silkworm/types/block_test.cpp
Comment (AndreaLanfranchi) [inline core/silkworm/types/block_test.cpp:194]: Eventually we can remove `boundary()` from header as now PoW check from ethash can check directly against difficulty

File: node/silkworm/db/access_layer.cpp
Comment (AndreaLanfranchi) [inline node/silkworm/db/access_layer.cpp:320]: Probably we need to check `data.key.length() >= kAddressLength + kHashLength`.
Should a wrong record be in table then substr would throw an out of bounds exception

File: node/silkworm/db/access_layer.cpp
Comment (AndreaLanfranchi) [inline node/silkworm/db/access_layer.cpp:399]: Maybe replace this with SILKWORM_ASSERT ?

File: node/silkworm/stagedsync/stage_tx_lookup.cpp
Comment (AndreaLanfranchi) [inline node/silkworm/stagedsync/stage_tx_lookup.cpp:158]: Since MDBX 0.10.5 this can be simplified as 
```suggestion
                lookup_table.erase(db::to_slice(hash.bytes));
```

**Review by AndreaLanfranchi — CHANGES_REQUESTED** [review]

File: node/silkworm/stagedsync/stage_tx_lookup.cpp
Comment (yperbasis) [reply node/silkworm/stagedsync/stage_tx_lookup.cpp]: Done in https://github.com/torquem-ch/silkworm/pull/494/commits/0234a0eee7d2db43a2489a407aa5478b5a798e22

File: node/silkworm/db/access_layer.cpp
Comment (yperbasis) [reply node/silkworm/db/access_layer.cpp]: Sure; done in https://github.com/torquem-ch/silkworm/pull/494/commits/ef07eea0b72edaf64b11ac1336b1af8b30b24e9a

File: node/silkworm/db/access_layer.cpp
Comment (yperbasis) [reply node/silkworm/db/access_layer.cpp:321]: Right; done in https://github.com/torquem-ch/silkworm/pull/494/commits/ef07eea0b72edaf64b11ac1336b1af8b30b24e9a

File: core/silkworm/common/util.cpp
Comment (yperbasis) [reply core/silkworm/common/util.cpp:126]: `ByteView` is a small type. Google also suggests to pass it by value: see "A Few Additional Notes" in https://abseil.io/tips/1.

File: core/silkworm/types/block_test.cpp
Comment (yperbasis) [reply core/silkworm/types/block_test.cpp:194]: Right. Let's do it in a separate PR since it's quite unrelated?

File: core/silkworm/common/util.cpp
Comment (AndreaLanfranchi) [reply core/silkworm/common/util.cpp:126]: I understand that but my point is made up of :
- even if small is neverheless the copy of a pointer **and** a size
- google says "should" ... not "must"
- google says we can keep consistency with surrounding code
Hence, imho, `const sv& data` does no harm and gives immediate picture the param is not consumed (which is consistent with other functions)

File: core/silkworm/types/block_test.cpp
Comment (AndreaLanfranchi) [reply core/silkworm/types/block_test.cpp:194]: Fair enough

**Review by AndreaLanfranchi — APPROVED** [review]

**Review by mriccobene — APPROVED** [review]

**Pull request merged**
